### PR TITLE
Upgrades

### PR DESCRIPTION
**Pull request by RoyEJohnson**
for: openstax/unified#2296

## Notes
`glob` could update to latest, but that requires a later version of node. The CI doesn't seem to respect .nvmrc.
`styled-components` and `styled-icons` could update, but that requires making screenshots.

### CODE REVIEW

File: package.json
Comment (TomWoodward) [inline package.json:46]: the change to the UUID package has been problematic for us because they changed their module format -- if it works its fine, but if you get strange issues check this

**Review by TomWoodward — APPROVED** [review]
the errors/warnings in the CI output is consistent with what is happening on main, this looks good

File: package.json
Comment (RoyEJohnson) [reply package.json:46]: v8 and v9 didn't work, but v7 was ok.

**Comment by RoyEJohnson** [comment]
@TomWoodward I'm scaling the PR back to just the Dev Dependencies for now. I'm guessing that would be the safest batch of changes. Recent deployments were crashing despite successful local lint/build/test/run, so I'm hoping this will get back to successful deployment.

**Comment by RoyEJohnson** [comment]
Apparently, when it tells you it was deployed, it takes some time before the deployment is actually functional? Things I am learning...

**Comment by RoyEJohnson** [comment]
@TomWoodward I have finished going through the outdated packages to determine what is the most recent version they can be upgraded to and still build successfully. `yarn upgrade outdated` is currently a no-op. :)
If you would prefer that I split the upgrades into multiple PRs, I can do that. Do you have a preferred grouping (e.g., dev dependencies in one, run dependencies in another)?

**Review by TomWoodward — CHANGES_REQUESTED** [review]
please double check that the types definition versions line up with the package versions, otherwise this looks fine

File: package.json
Comment (TomWoodward) [reply package.json:119]: the patch versions aren't going to line up, but the major/minor should 

https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library

**Comment by TomWoodward** [comment]
re 
>  The CI doesn't seem to respect .nvmrc.

dante recently fixed this in the other repos, you could pull in the change if it helps https://github.com/openstax/assignments/pull/25/files

**Comment by TomWoodward** [comment]
also i'm never quite sure if you're still working on this or waiting for my review, let me know when i should take a look

**Comment by RoyEJohnson** [comment]
I think this is ready to review. I've used ~#.# for most of the @types and corresponding packages, to document the expected major-minor matching. Some packages were out of version sync already, and they may have stayed out of sync if it was the best I could do; the types package is 2 minor revisions behind in those case.
We'll leave upgrading node for another day. This is, as you noted, quite big enough as it is.

**Comment by TomWoodward** [comment]
idk why this is working in CI but not locally, when i try to run the `yarn prerender:local` i get `Error: Cannot find module '@aws-sdk/util-base64-node'` have you seen that?

**Comment by RoyEJohnson** [comment]
I do not have the issue with `REACT_APP_ENV=test yarn prerender:local`. It indicates success. You ran `yarn` prior?

**Comment by RoyEJohnson** [comment]
I don't have that module installed, and I don't know what would be trying to use it.
```
% yarn why @aws-sdk/util-base64-node      
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "@aws-sdk/util-base64-node"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
error We couldn't find a match!
✨  Done in 1.04s.
```

**Comment by TomWoodward** [comment]
the stacktrace looks like:

```
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/client-sts/dist-cjs/runtimeConfig.js                                              
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/client-sts/dist-cjs/STSClient.js                                                  
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/client-sts/dist-cjs/STS.js                                                        
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/client-sts/dist-cjs/index.js                                                      
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/dist-cjs/runtimeConfig.js                                                                               
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/dist-cjs/S3Client.js                                                                                    
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/dist-cjs/S3.js                                                                                          
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/dist-cjs/index.js                                                                                       
- /Users/tomwoodward/code/openstax/books-web/script/prerender/fileUtils.ts
- /Users/tomwoodward/code/openstax/books-web/script/prerender/contentPages.tsx
- /Users/tomwoodward/code/openstax/books-web/script/prerender/local.ts
- /Users/tomwoodward/code/openstax/books-web/script/entry.js
```

i also have 
```
[1602.43][upgrades][b5f224fc7]~/code/openstax/books-web: yarn why @aws-sdk/client-s3
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "@aws-sdk/client-s3"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "@aws-sdk/client-s3@3.297.0"
info Has been hoisted to "@aws-sdk/client-s3"
info This module exists because it's specified in "devDependencies".
info Disk size without dependencies: "6.77MB"
info Disk size with unique dependencies: "21.56MB"
info Disk size with transitive dependencies: "32.07MB"
info Number of shared dependencies: 85
✨  Done in 1.32s.

[1602.43][upgrades][b5f224fc7]~/code/openstax/books-web: yarn why @aws-sdk/util-base64-node
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "@aws-sdk/util-base64-node"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
error We couldn't find a match!
✨  Done in 1.01s.

[1602.43][upgrades][b5f224fc7]~/code/openstax/books-web: yarn why @aws-sdk/client-sts
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "@aws-sdk/client-sts"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "@aws-sdk/client-sts@3.297.0"
info Has been hoisted to "@aws-sdk/client-sts"
info Reasons this module exists
   - Hoisted from "@aws-sdk#client-cloudformation#@aws-sdk#client-sts"
   - Hoisted from "@aws-sdk#client-s3#@aws-sdk#client-sts"
   - Hoisted from "@aws-sdk#client-sqs#@aws-sdk#client-sts"
   - Hoisted from "@aws-sdk#credential-providers#@aws-sdk#client-sts"
   - Hoisted from "@aws-sdk#credential-providers#@aws-sdk#client-cognito-identity#@aws-sdk#client-sts"
info Disk size without dependencies: "840KB"
info Disk size with unique dependencies: "11.56MB"
info Disk size with transitive dependencies: "19.81MB"
info Number of shared dependencies: 66
=> Found "@openstax/ts-utils#@aws-sdk/client-sts@3.235.0"
info This module exists because "@openstax#ts-utils" depends on it.
info Disk size without dependencies: "812KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "20.06MB"
info Number of shared dependencies: 67
=> Found "@aws-sdk/client-dynamodb#@aws-sdk/client-sts@3.235.0"
info This module exists because "@openstax#ts-utils#@aws-sdk#client-dynamodb" depends on it.
info Disk size without dependencies: "812KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "20.06MB"
info Number of shared dependencies: 67
=> Found "@aws-sdk/client-ssm#@aws-sdk/client-sts@3.235.0"
info This module exists because "@openstax#ts-utils#@aws-sdk#client-ssm" depends on it.
info Disk size without dependencies: "812KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "20.06MB"
info Number of shared dependencies: 67
✨  Done in 1.91s.
```

does any of that look different from you? i might have an older dependency that isn't locked or something.

**Comment by TomWoodward** [comment]
i'm sure that something weird is happening here, but if i delete my node_modules and reinstall everything works. other developers may also have to do the same, but i'm tired of trying to figure it out so maybe we should just merge

**Review by TomWoodward — APPROVED** [review]

**Pull request merged**
